crates/storage/src/test_utils.rs

```diff
@@ -169,8 +169,13 @@ pub(crate) fn extract_events(
         .collect()
 }
 
-/// Creates a storage instance in memory with a set of expected emitted events
-pub fn setup_test_storage() -> (Storage, Vec<StarknetEmittedEvent>) {
+pub fn setup_test_storage_and_return_test_data() -> (
```

Comment:
Might be a good idea to refactor this so that its just a single test setup method?

I imagine it could instead return something like `(Storage, TestData)` with
```rust
struct TestData {
    blocks: Vec<StarknetBlock>,
    ...
}
```
This would also help future proof as we can just extend `TestData` without disturbing existing tests.

(I do understand that this could be a pita).

Reply:
Ah, that's a good idea. I'll check how much work this would be.

Reply:
@Mirko-von-Leipzig I've changed the test setup code to what you've suggested.

Reply:
I think it turned out great 🦾 We should really do similar things elsewhere to improve our ability to simplify test code.